### PR TITLE
Azsearch retro feedback

### PR DESCRIPTION
**Pull request by CodeRunRepeat**
Feedback from the Azure Search Java SDK project retrospective:

1. Planning user feedback: In each epic/feature, include a user story about collecting external feedback.
2. Working agreement on reviews: At the beginning of the project, we extend the Working Agreement to include:
  * How and when we do design reviews
  * Where all the design decisions are documented
3. Planning design reviews: Each user story includes a design review task; during planning this is activated or deactivated.

### CODE REVIEW

File: TECH-LEADS-CHECKLIST.md [hunk 1]
```diff
@@ -1,71 +1,85 @@
...
 - [ ] Scrum Master (fixed/rotating) to run standup daily.
 - [ ] Agile process clearly defined within team.
-- [ ] Tech Lead (+ PO/Others) have responsability for backlog management and grooming.
+- [ ] Tech Lead (+ PO/Others) have responsibility for backlog management and grooming.
 - [ ] Working agreement between members of team and customer.
 
 ## Design Reviews
+
 - [ ] Process for conducting design reviews.
+- [ ] Including design review process in the [Working Agreement](/team-agreements/working-agreements/readme.md)
```
Comment (fnocera) [inline TECH-LEADS-CHECKLIST.md]: Would perhaps be better to include in the first bullet (above) eg: "Process for conducting design reviews is documented in [Working Agreement](/team-agreements/working-agreements/readme.md)

File: TECH-LEADS-CHECKLIST.md [hunk 2]
```diff
@@ -1,71 +1,85 @@
...
 - [ ] Design reviews for each major component of the solution.
-- [ ] Document major design decisions and alternatives. 
-- [ ] Stories and/or PRs link to the design document
+- [ ] Document major design decisions and alternatives.
```
Comment (fnocera) [inline TECH-LEADS-CHECKLIST.md]: Again as above combine this one and the above bullet eg: "Design reviews are carried out and documented for each major component of the solution."

File: source-control/readme.md [hunk 3]
```diff
@@ -3,17 +3,25 @@
...
 ## Goal
+
 * Following industry best practice to work in geo-distributed teams which encourage contributions from all across [CSE](../CSE.md) as well as the broader OSS community
 * Improve code quality by enforcing reviews before merging into master branches
 * Improve traceability of features and fixes through a clean commit history
 
 ## General Guidance
-Consistency is important, so agree to the approach as a team before starting to code. The team should at least be doing the following:
+
+Consistency is important, so agree to the approach as a team before starting to code. Consider this as a design decision
```
Comment (fnocera) [inline source-control/readme.md]: Some re-wording suggestions to make it clearer and more concise "Consistency is important, so agree on the approach as a team before starting to code. Treat this as a design decision, so include a design proposal and review, in the same way as you would document all design decisions (see [Working Agreements](../team-agreements/working-agreements/readme.md))."

File: team-roles/readme.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+TODO: link to role descriptions
+
+The Scrum Master can also be accountable for PR turnaround, assigning reviewers and following up with them to make sure that reviews are done in a timely fashion.
+
+## Other roles
+
+On top of the roles defined in the Scrum process, defining one or some of the virtual roles below can
+help the team navigate specific situations successfully:
+
```
Comment (fnocera) [inline team-roles/readme.md]: Have these roles been discussed with the Agile Engineering Fundamentals Champions? I would want to make sure they are in agreement as I have personally not worked on a project with these roles. 
We usually absorb responsibility for guidelines to everyone on the team and ultimate responsability to the Tech Lead and co-lead on the customer side. For the "Merge master" we leave it up to the person who owns the PR to complete their PRs once they have addressed all comments etc, good CI/CD and testing usually helps us manage most of this.

File: team-roles/readme.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# Team roles
```
Comment (fnocera) [inline team-roles/readme.md]: I would consider moving the team-roles section into team-agreements. What do you think?

File: TECH-LEADS-CHECKLIST.md (hunk 1 above)
Comment (CodeRunRepeat) [reply TECH-LEADS-CHECKLIST.md]: Updated.

File: TECH-LEADS-CHECKLIST.md (hunk 2 above)
Comment (CodeRunRepeat) [reply TECH-LEADS-CHECKLIST.md]: Updated.

File: source-control/readme.md (hunk 3 above)
Comment (CodeRunRepeat) [reply source-control/readme.md]: Updated.

**Comment by CodeRunRepeat** [comment]
> 
> 
> Looks good, but it looks like you forgot to add link to role description.
> This one can be good: https://www.scrum-institute.org/Scrum_Roles_The_Scrum_Team.php
> I think you can add a 1-liner per each role, and link to this page for further reading.

Good feedback @limorl . As per Federica's comments, I will retract the team roles content and discuss with the champions first before resubmitting; if it is added then I will include the links.

File: team-roles/readme.md (hunk 4 above)
Comment (CodeRunRepeat) [reply team-roles/readme.md]: @fnocera I will remove this file for the time being and discuss with the champions first; the suggestion to consider these additional roles comes from real life experiences and successful practices so I'd argue that the approach you describe above may vary. Thanks for the feedback.

File: team-roles/readme.md (hunk 5 above)
Comment (CodeRunRepeat) [reply team-roles/readme.md]: File was removed.

**Comment by CodeRunRepeat** [comment]
> Thank you for this awesome contribution :) - think there are a few things that might need to be edited/moved around a little and I think we need more time to consider the team-roles addition, perhaps remove the team roles for now and then following more discussion we can decide whether to add back in?

@fnocera team roles have been removed. Please confirm.

**Pull request merged**